### DISCUSSION TITLE
Globally convert ValidationError to a custom error

### DISCUSSION THREAD

**Discussion by amh4r** [pydantic/pydantic#7902] [Question] [answered]
I'd like to use Pydantic to validate config options for my library, but I'd like to avoid exposing Pydantic internals to my users. Instead, I'd like to catch each `ValidationError`, parse it, and then reraise my own `InvalidConfig` error. How do I write a function that globally converts all `ValidationError`s?

For example, users could configure my library like this:
```py
import inngest

@inngest.create_function(
    inngest.FunctionOpts(id="my_function"),
)
def fn() -> str:
    return "Hello"
```

Where `FunctionOpts` extends `pydantic.BaseModel`.

But when my users write that code, any invalid `FunctionOpts` data results in a `pydantic.ValidationError` being raised.

Thanks for the help!

**Comment by amh4r** [comment, marked as answer]
Nevermind! The solution is easy:

```py
class _BaseOpts(BaseModel):
    def __init__(self, *args: object, **kwargs: object) -> None:
        try:
            super().__init__(*args, **kwargs)
        except ValidationError as err:
            raise MyError()

class FunctionOpts(_BaseOpts):
    id: str
```

**Comment by KonstantinKlepikov** [comment]
@amh4r so, now in p2 is actualy:

```python
class _BaseOpts(BaseModel):

    def __init__(self, /, **data: Any) -> None:
        try:
            super().__init__(**data)
        except ValidationError as err:
            raise MyError()
```
